apps/api/src/domain/domainModelValidators/bibliographicReferenceValidator/index.ts

```diff
@@ -1,11 +1,36 @@
 import { InternalError } from '../../../lib/errors/InternalError';
+import BookBibliographicReferenceData from '../../models/bibliographic-reference/entities/BookBibliographicReferenceData';
+import { IBibliographicReference } from '../../models/bibliographic-reference/interfaces/IBibliographicReference';
+import { BibliographicReferenceType } from '../../models/bibliographic-reference/types/BibliographicReferenceType';
+import { EntityId } from '../../types/ResourceId';
+import { resourceTypes } from '../../types/resourceTypes';
+import { isNullOrUndefined } from '../../utilities/validation/is-null-or-undefined';
+import InvalidEntityDTOError from '../errors/InvalidEntityDTOError';
 import { DomainModelValidator } from '../types/DomainModelValidator';
+import buildSimpleDiscriminatedUnionValidationFunction from '../utilities/buildSimpleDiscriminatedUnionValidationFunction';
 import { Valid } from '../Valid';
 
+const buildTopLevelError = (id: EntityId, innerErrors: InternalError[]): InternalError =>
+    new InvalidEntityDTOError(resourceTypes.bibliographicReference, id, innerErrors);
+
 const bibliographicReferenceValidator: DomainModelValidator = (
-    _: unknown
+    dto: unknown
 ): Valid | InternalError => {
-    // TODO implement me and add a test case.
+    if (isNullOrUndefined(dto))
+        return new InvalidEntityDTOError(resourceTypes.bibliographicReference);
+
+    const allErrors: InternalError[] = [];
+
+    const { data, id } = dto as IBibliographicReference;
+
+    const dataValidationFunction = buildSimpleDiscriminatedUnionValidationFunction([
```

Comment:
I wonder if we can just get away with calling `validateSimpleInvariants` as the entire implementation of this validator?